src/nvim/statusline.c

```diff
@@ -1804,5 +1811,14 @@ int build_stl_str_hl(win_T *wp, char *out, size_t outlen, char *fmt, int use_san
     curwin->w_redr_type = save_redr_type;
   }
 
+  // Check for an error.  If there is one we would loop in redrawing the screen.
+  // Avoid that by making the corresponding option empty. Pass OPT_LOCAL for
+  // 'numberformat' and 'statusline' if appropriate.
+  if (did_emsg != save_did_emsg) {
+    int scope = *opt_name == 'n' || (*opt_name == 's' && *wp->w_p_stl != NUL)
```

Comment:
`*opt_name == 'n'/'s'` seems fragile, might want to `assert(*opt_name != 'n' || opt_name != "numberformat")` somewhere early in the codepath outside of the hotloop.

Reply:
Sorry, not sure I understand what purpose the assert macro serves. I just thought we could avoid `strcmp()` here since all of the statusline options passed to this function start with different letters (for now). Could you make a suggest diff with the change you want?

Reply:
The purpose is to avoid accidental bugs if a new `opt` is used which also starts with "n" or "s".